src/nvs.rs

```diff
@@ -452,7 +453,10 @@ impl<T: NvsPartitionId> EspNvs<T> {
 
     pub fn set_str(&mut self, name: &str, val: &str) -> Result<(), EspError> {
         let c_key = CString::new(name).unwrap();
-        let c_val = CString::new(val).unwrap();
+        let c_val = match CString::new(val) {
```

Comment:
OK so you've addressed *this* `CString` construction, but you have not addressed the one above (for the key), nor the `CString` constructions elsewhere in this module. I think we should either address all `CString` constructions from user-provided `str`s - at least within this module - or none.

Reply:
I was just addressing the issue and I was not sure if the solution was acceptable for you. No problem, I'll complete the module.